### ISSUE TITLE
Kernel 4.8 Support?

### ISSUE THREAD

**Issue by matan129** [imageguy/fglrx-for-Fedora#8]
Hey,
I'm running Linux Kernel version `4.8.0-22-generic x86_64`, and I was wondering - do you plan to support it?

I tried using the 4.7 patch. 
The kernel module (`fglrx`) compiles successfully, but crashes when trying to use it. For example:

```
 WARNING: CPU: 0 PID: 26566 at /build/linux-NNryke/linux-4.8.0/drivers/pci/msi.c:1051 __pci_enable_msi_range+0x2bc/0x320
 Modules linked in: fglrx(POE) cpuid usblp dm_crypt intel_rapl x86_pkg_temp_thermal intel_powerclamp coretemp kvm_intel kvm irqbypass snd_hda_codec_via snd_hda_c
odec_hdmi snd_hda_codec_generic snd_usb_audio input_leds snd_hda_intel crct10dif_pclmul crc32_pclmul ghash_clmulni_intel aesni_intel joydev hid_waltop snd_usbmidi_lib aes_x86_64 lrw snd_hda_codec glue_helpe
r snd_hda_core ablk_helper snd_hwdep cryptd snd_pcm intel_cstate intel_rapl_perf serio_raw snd_seq_midi snd_seq_midi_event snd_rawmidi snd_seq snd_seq_device snd_timer snd soundcore shpchp mei_me mei ie3120
0_edac lpc_ich edac_core mac_hid tpm_infineon binfmt_misc ip6t_REJECT nf_reject_ipv6 nf_log_ipv6 xt_hl ip6t_rt nf_conntrack_ipv6 nf_defrag_ipv6 ipt_REJECT nf_reject_ipv4 nf_log_ipv4 nf_log_common xt_LOG xt_
limit xt_tcpudp xt_addrtype nf_conntrack_ipv4 nf_defrag_ipv4 xt_conntrack ip6table_filter ip6_tables nf_conntrack_netbios_ns nf_conntrack_broadcast nf_nat_ftp i915 nf_nat nf_conntrack_ftp nf_conntrack iptab
le_filter parport_pc ppdev lp parport ip_tables x_tables autofs4 hid_generic ses enclosure scsi_transport_sas hid_microsoft usbhid hid uas usb_storage amdkfd amd_iommu_v2 mxm_wmi radeon i2c_algo_bit ttm drm
_kms_helper syscopyarea psmouse sysfillrect ahci sysimgblt fb_sys_fops libahci drm atl1c video fjes wmi
CPU: 0 PID: 26566 Comm: ipython Tainted: P           OE   4.8.0-22-generic #24-Ubuntu
Hardware name: Gigabyte Technology Co., Ltd. To be filled by O.E.M./Z77MX-D3H, BIOS F15 09/17/2012
 0000000000000286 0000000066bd2d5f ffff92d539773918 ffffffff9ee2f5d2
 0000000000000000 0000000000000000 ffff92d539773958 ffffffff9ea82e5b
 0000041bc0ebc748 0000000000000001 ffff92d539773a40 0000000000000001
Call Trace:
 [<ffffffff9ee2f5d2>] dump_stack+0x63/0x81
 [<ffffffff9ea82e5b>] __warn+0xcb/0xf0
 [<ffffffff9ea82f8d>] warn_slowpath_null+0x1d/0x20
 [<ffffffff9eea151c>] __pci_enable_msi_range+0x2bc/0x320
 [<ffffffffc0bf9065>] ? fireglPcsCommand+0xa5/0x140 [fglrx]
 [<ffffffff9eea1590>] pci_enable_msi_range+0x10/0x20
 [<ffffffffc0be4688>] KCL_RequestMSI+0x18/0x30 [fglrx]
 [<ffffffffc0c082dc>] IRQMGR_initialize+0x2bc/0x410 [fglrx]
 [<ffffffffc0c07fc7>] ? IRQMGR_alloc_context+0xb7/0x110 [fglrx]
 [<ffffffffc0c06807>] ? irqmgr_wrap_initialize+0x27/0x140 [fglrx]
 [<ffffffffc0c07dc5>] ? firegl_irqmgr_init+0x55/0x100 [fglrx]
 [<ffffffffc0c19111>] ? hal_init_gpu+0x221/0x4e0 [fglrx]
 [<ffffffffc0bf257e>] ? firegl_open+0x2de/0x310 [fglrx]
 [<ffffffffc0be0277>] ? ip_firegl_open+0x17/0x20 [fglrx]
 [<ffffffffc0be1b06>] ? firegl_stub_open+0x96/0x100 [fglrx]
 [<ffffffff9ec37b9c>] ? chrdev_open+0xbc/0x1a0
 [<ffffffff9ec2ff55>] ? do_dentry_open+0x205/0x300
 [<ffffffff9ec37ae0>] ? cdev_put+0x30/0x30
 [<ffffffff9ec310ec>] ? vfs_open+0x4c/0x70
 [<ffffffff9ec3e24b>] ? may_open+0x9b/0x100
 [<ffffffff9ec42364>] ? path_openat+0x664/0x1450
 [<ffffffffc0be2ee6>] ? KCL_SPINLOCK_STATIC_Grab+0x16/0x20 [fglrx]
 [<ffffffffc0beac2b>] ? drm_free+0x11b/0x1a0 [fglrx]
 [<ffffffffc0be88f2>] ? KCL_DEBUG_Print_Trace+0x32/0xf0 [fglrx]
 [<ffffffff9ec44491>] ? do_filp_open+0x91/0x100
 [<ffffffff9ec52ea6>] ? __alloc_fd+0x46/0x180
 [<ffffffff9ec31505>] ? do_sys_open+0x135/0x280
 [<ffffffff9ec3166e>] ? SyS_open+0x1e/0x20
 [<ffffffff9f29f076>] ? entry_SYSCALL_64_fastpath+0x1e/0xa8
```

In the above log (from `kern.log`), I was trying to load some OpenCL library. Also, when I starting the GUI it hangs and turns to black (with similar kernel messages).

**Comment by imageguy** [comment]
Hi,

I was out for a few days (just got back) and haven't tried 4.8 yet. I am 
using fglrx on my main box at home, so I will get it running if I can at 
all. I use _.fc_.x86_64 kernels since I run Fedora distro, so I don't 
have a way to try generic. I checked the main discussion page at 
https://bluehatrecord.wordpress.com/2016/03/25/installing-the-proprietary-amd-crimson-driver-on-fedora-23-with-linux-kernel-4-4-6-300/ 
and there is a comment at the bottom from Oct 17 with someone having a 
similar problem on Fedora 25 Beta. So I'd bet there is a general 4.8. 
issue, not tied to fc24 vs. generic.

I will admit, though, that I've had persistent OpenCL problem - no login 
screen shown, though when I press "enter", put in my password and press 
enter again, I do log in. There is a supposed workround for this, but I 
haven't been able to make it work.

I'll give it a shot tomorrow and let you know how it turns out.

Regards, Nenad

**Comment by imageguy** [comment]
Fedora 24 hasn't shipped 4.8 kernel yet - I updated today and got 4.7.7-200.fc24.x86_64. I run stock Fedora, so it'll have to wait until fc24 ships 4.8.....

**Comment by matan129** [comment]
Okay, thanks for the response. It was a longshot anyway. I hope that the open source drivers will [catch up](https://dri.freedesktop.org/wiki/GalliumCompute/) in the near future.
